include/finufft/finufft_core.h

```diff
@@ -151,30 +149,26 @@ template<typename TF> struct FINUFFT_PLAN_T { // the main plan class, fully C++
   FINUFFT_PLAN_T &operator=(const FINUFFT_PLAN_T &) = delete;
   ~FINUFFT_PLAN_T();
 
-  int type;                // transform type (Rokhlin naming): 1,2 or 3
-  int dim;                 // overall dimension: 1,2 or 3
-  int ntrans;              // how many transforms to do at once (vector or "many" mode)
-  BIGINT nj;               // num of NU pts in type 1,2 (for type 3, num input x pts)
-  BIGINT nk;               // number of NU freq pts (type 3 only)
-  TF tol;                  // relative user tolerance
-  int batchSize;           // # strength vectors to group together for FFTW, etc
-  int nbatch;              // how many batches done to cover all ntrans vectors
+  int type;      // transform type (Rokhlin naming): 1,2 or 3
+  int dim;       // overall dimension: 1,2 or 3
+  int ntrans;    // how many transforms to do at once (vector or "many" mode)
+  BIGINT nj;     // num of NU pts in type 1,2 (for type 3, num input x pts)
+  BIGINT nk;     // number of NU freq pts (type 3 only)
+  TF tol;        // relative user tolerance
+  int batchSize; // # strength vectors to group together for FFTW, etc
+  int nbatch;    // how many batches done to cover all ntrans vectors
 
-  BIGINT ms;               // number of modes in x (1) dir (historical CMCL name) = N1
-  BIGINT mt;               // number of modes in y (2) direction = N2
-  BIGINT mu;               // number of modes in z (3) direction = N3
-  BIGINT N;                // total # modes (prod of above three)
+  std::array<UBIGINT, 3> mstu; // number of modes in x/y/z dir (historical CMCL name) =
+                               // N1/N2/N3
+  UBIGINT N;                   // total # modes (prod of above three)
 
-  BIGINT nf1 = 1;          // size of internal fine grid in x (1) direction
-  BIGINT nf2 = 1;          // " y (2)
-  BIGINT nf3 = 1;          // " z (3)
-  BIGINT nf  = 1;          // total # fine grid points (product of the above three)
+  std::array<UBIGINT, 3> nf123 = {1, 1, 1}; // size of internal fine grid in x/y/z
```

Comment:
the = constructor might be redundant nf123{1,1,1}

Reply:
Fine!